app/src/main/java/io/appium/uiautomator2/server/ServerInstrumentation.java

```diff
@@ -73,20 +76,39 @@ private void setAccessibilityServiceState() {
         }
     }
 
-    private ServerInstrumentation(Context context, int serverPort) {
+    private ServerInstrumentation(Context context, int serverPort, int mjpegServerPort) {
         if (!isValidPort(serverPort)) {
             throw new UiAutomator2Exception(String.format(
-                    "The port is out of valid range [%s;%s]: %s", MIN_PORT, MAX_PORT, serverPort));
+                "The server port is out of valid range [%s;%s]: %s",
+                MIN_PORT,
+                MAX_PORT,
+                serverPort
+            ));
         }
         this.serverPort = serverPort;
+
+        if (!isValidPort(mjpegServerPort)) {
+            throw new UiAutomator2Exception(String.format(
```

Comment:
I'd rather simply log a warning here if we want to make mjpeg server less intrusive

Reply:
Good call, this will be resolved w/ the upcoming update(s)